mne/bem.py

```diff
@@ -332,3 +337,116 @@ def cost_fun(x, points):
+                       verbose=None):
+    """
+    Create BEM surfaces using the watershed algorithm included with FreeSurfer
+
+    Parameters
+    ----------
+    subject : str
+        Subject name (required)
+    subjects_dir : str
+        Directory containing subjects data. If None use
+        the Freesurfer SUBJECTS_DIR environment variable.
+    overwrite : bool
+        Write over existing files
+    volume : str
+        Defaults to T1
+    atlas : bool
+        Specify the --atlas option for mri_watershed
+    gcaatlas : bool
+        Use the subcortical atlas
+    preflood : int
+        Change the preflood height
+    verbose : bool, str or None
+        If not None, override default verbose level
+
+    .. versionadded:: 0.10
+    """
+    env = os.environ.copy()
+
+    if not os.environ.get('FREESURFER_HOME'):
+        raise RuntimeError('FREESURFER_HOME environment variable not set')
+
+    env['SUBJECT'] = subject
+
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+    env['SUBJECTS_DIR'] = subjects_dir
+
+    subject_dir = op.join(subjects_dir, subject)
+    mri_dir = op.join(subject_dir, 'mri')
+    T1_dir = op.join(mri_dir, volume)
+    T1_mgz = op.join(mri_dir, volume + '.mgz')
+    bem_dir = op.join(subject_dir, 'bem')
+    ws_dir = op.join(subject_dir, 'bem', 'watershed')
+
+    if not op.exists(subject_dir):
+        raise RuntimeError('Could not find the MRI data directory "%s"'
+                           % subject_dir)
+    if not op.exists(bem_dir):
+        os.makedirs(bem_dir)
+    if not op.exists(T1_dir) and not op.exists(T1_mgz):
+        raise RuntimeError('Could not find the MRI data')
+    if op.exists(ws_dir):
+        if not overwrite:
+            raise RuntimeError('%s already exists. Use the --overwrite option'
+                               'to recreate it.' % ws_dir)
+        else:
+            shutil.rmtree(ws_dir)
+    # put together the command
+    cmd = ['mri_watershed']
+    if preflood:
+        cmd += ["-h",  "%s" % int(preflood)]
+
+    if gcaatlas:
+        cmd += ['-atlas', '-T1', '-brain_atlas', env['FREESURFER_HOME'] +
+                '/average/RB_all_withskull_2007-08-08.gca',
+                subject_dir + '/mri/transforms/talairach_with_skull.lta']
+    elif atlas:
+        cmd += ['-atlas']
+    if op.exists(T1_mgz):
+        cmd += ['-useSRAS', '-surf', op.join(ws_dir, subject), T1_mgz,
+                op.join(ws_dir, 'ws')]
+    else:
+        cmd += ['-useSRAS', '-surf', op.join(ws_dir, subject), T1_dir,
+                op.join(ws_dir, 'ws')]
+    # report and run
+    logger.info('\nRunning mri_watershed for BEM segmentation with the '
+                'following parameters:\n\n'
+                'SUBJECTS_DIR = %s\n'
+                'SUBJECT = %s\n'
+                'Result dir = %s\n' % (subjects_dir, subject, ws_dir))
+    os.makedirs(op.join(ws_dir, 'ws'))
+    run_subprocess(cmd, env=env, stdout=sys.stdout)
```

Comment:
note. it would be nicer if run_subprocess printed the entire command, not a list of strings. It would make copying and pasting easier, e.g. for debugging in the command line.